src/main/java/nextstep/subway/line/application/LineService.java

```diff
@@ -75,108 +57,24 @@ public void deleteLineById(Long id) {
 
     public void addLineStation(Long lineId, SectionRequest request) {
         Line line = findLineById(lineId);
-        Station upStation = stationService.findStationById(request.getUpStationId());
-        Station downStation = stationService.findStationById(request.getDownStationId());
-        List<Station> stations = getStations(line);
-        boolean isUpStationExisted = stations.stream().anyMatch(it -> it == upStation);
-        boolean isDownStationExisted = stations.stream().anyMatch(it -> it == downStation);
-
-        if (isUpStationExisted && isDownStationExisted) {
-            throw new RuntimeException("이미 등록된 구간 입니다.");
-        }
-
-        if (!stations.isEmpty() && stations.stream().noneMatch(it -> it == upStation) &&
-                stations.stream().noneMatch(it -> it == downStation)) {
-            throw new RuntimeException("등록할 수 없는 구간 입니다.");
-        }
-
-        if (stations.isEmpty()) {
-            line.getSections().add(new Section(line, upStation, downStation, request.getDistance()));
-            return;
-        }
-
-        if (isUpStationExisted) {
-            line.getSections().stream()
-                    .filter(it -> it.getUpStation() == upStation)
-                    .findFirst()
-                    .ifPresent(it -> it.updateUpStation(downStation, request.getDistance()));
-
-            line.getSections().add(new Section(line, upStation, downStation, request.getDistance()));
-        } else if (isDownStationExisted) {
-            line.getSections().stream()
-                    .filter(it -> it.getDownStation() == downStation)
-                    .findFirst()
-                    .ifPresent(it -> it.updateDownStation(upStation, request.getDistance()));
-
-            line.getSections().add(new Section(line, upStation, downStation, request.getDistance()));
-        } else {
-            throw new RuntimeException();
-        }
+        Station upStation = findStationById(request.getUpStationId());
+        Station downStation = findStationById(request.getDownStationId());
+
+        line.addLineStation(upStation, downStation, request.getDistance());
```

Comment:
Section을 서비스에서 만들어보는건 어떨까요? 🤔 
line에게 section을 생성할 책임을 가지기는 조금 무거워보이기도 하는 것 같아요 :)
개인적으로 파라미터가 많아지는게 하나의 신호이지 않을까 하는 생각도 드네요 !
```suggestion
        Section section = new Section(line, upStation, downStation, Distance.of(request.getDistance()));
        line.addSection(section);
```

Reply:
좋은 방법인 것 같습니다!!

Reply:
로직을 도메인으로 옮기면서 도메인에서 정합성 체크나 기존 구간 길이 재계산 등을 하고 있습니다. 
Section을 생성하는 건 서비스에서 한다고 하더라고 인수가 모두 가야하는 건 똑같은거 같은데..
좋은 방법이 있을까요??

Reply:
서비스에서 Section에 대한 생성을 해주도록 하게 되면, line의 addSection의 인자는 하나로 줄어드는데요. 생성자의 인자의 수가 그대로이신 부분을 이야기해주시는 부분일까요 ?